src/api/spec/models/bs_request_action_spec.rb

```diff
@@ -360,4 +360,38 @@
       end
     end
   end
+
+  describe '#check_for_expand_errors' do
+    let(:project) { user.home_project }
+    let(:target_package) do
+      create(:package_with_file, name: 'package_encoding_1', project: project)
+    end
+    let(:source_package) do
+      create(:package_with_file, name: 'package_encoding_2', project: project)
+    end
+    let(:bs_request) { create(:bs_request_with_submit_action, creator: user, target_package: target_package, source_package: source_package) }
+    let(:bs_request_action) { bs_request.bs_request_actions.first }
+
+    context 'when there is a problem with the backend' do
+      context 'and we have a revision' do
+        before { bs_request_action.source_rev = '1234' }
+
+        it 'returns an error message' do
+          mocked_backend = class_double('Backend::Api::Sources::Package')
```

Comment:
Mocking the backend call should be extracted in a `before` block under the `'when there is a problem with the backend'` context. This way, it doesn't have to be mocked in two examples.